src/filters.hpp

```diff
@@ -10,9 +10,13 @@
 class Filters : public Nan::ObjectWrap {
   public:
     using filter_value_type = mbgl::style::Filter;
+    using filter_properties = struct {
```

Comment:
I recommend:

 -  calling this `filter_properties_type`. 
  - making this a `std::pair` instead of a custom `struct` (more idiomatic in C++)
  - having the `type` be an enum for faster comparisons (rather than string)

Reply:
done in the next commit:
* change the name
* use `std::pair`
* change the type into `enum`

Reply:
Looks great. Nice work.